packages/api/test/functional/permission.spec.js

```diff
@@ -0,0 +1,99 @@
+const { test, trait } = use('Test/Suite')('Permission');
+trait('Test/ApiClient');
+trait('Auth/Client');
+trait('DatabaseTransactions');
+
+const Permission = use('App/Models/Permission');
+const User = use('App/Models/User');
+
+const permission = {
+	permission: 'TEST_PERMISSION',
+	description: 'Test Permission',
+};
+
+const user = {
+	username: 'sabiatestinguser',
+	email: 'sabiatestingemail@gmail.com',
+	password: '123123',
+};
+
+test('try to access resource without authorization', async ({ client }) => {
+	await Permission.create(permission);
+
+	const response = await client.get('/permissions').end();
+
+	response.assertStatus(401);
+});
+
+test('GET /permissions Show a list of all permissions.', async ({ client }) => {
+	await Permission.create(permission);
+
+	const loggeduser = await User.create(user);
+
+	const response = await client
+		.get('/permissions')
+		.loginVia(loggeduser, 'jwt')
+		.end();
+
+	response.assertStatus(200);
+	response.assertJSONSubset([permission]);
+});
+
+test('POST /permissions create/save a new permission.', async ({ client }) => {
+	const loggeduser = await User.create(user);
+
+	const response = await client
+		.post('/permissions')
+		.send(permission)
+		.loginVia(loggeduser, 'jwt')
+		.end();
+
+	response.assertStatus(200);
```

Comment:
Para que tenhamos certeza, faça uma consulta ao banco usando o model para ter certeza que `permission` foi cadastrado no banco. Isso testa do início ao fim e faz com que não precisamos testar o model diretamente. 

Caso tenha um bug que não esteja salvando no banco mas faça com que o controller retorna status `200` esse teste irá passar.